content/docs/rfcs/12/README.md

```diff
@@ -16,7 +16,9 @@ contributors:
 
 # Content filtering
 
-**Protocol identifier***: `/vac/waku/filter/2.0.0-beta1`
+**Protocol identifiers**:
+- _filter-subscribe_: `/vac/waku/filter-subscribe/2.0.0-beta1`
+- _filter-push_: `/vac/waku/filter-push/2.0.0-beta1`
```

Comment:
I don't finish to see why we need to split the protocol ID into two different protocols 🤔 In my head:

> The filter protocol acts as a bidirectional protocol with a control path and a data path (like libp2p's Gossipsub has): 
> 
> * **The control path:** The gateway acts as a server for the "request-response" subscription management flow. The subscriber acts as the client requesting subscriptions.
> * **The data path:** The gateway delivers (via "push") the messages matching the subscription subscribers.

Based on that, it makes sense to announce (via libp2p's identify protocol) the filter protocol ID. A subscriber's peer managers will use the protocol ID to list this node as capable of managing filter subscriptions and pushing the subscriptions.

But what is the point of having a separate "receiver" protocol ID?

Reply:
If I understand you correctly, this is because the "data path" requires a connection/stream being negotiated _from_ the filter service node _to_ the client who requested the subscription. This requires the _client_ to have some protocol mounted which can be negotiated for this stream (where it is acting like a "service"). If this is simply the same as the subscribe protocol ID, the subscriber's peer manager will incorrectly list other subscribers (not capable of managing subscriptions/pushing) as possible service nodes. Not sure if this is what you meant?

I agree re similarities to gossipsub, but note that this is not currently a true "bidirectional" protocol. It does negotiate streams/connections in two directions, but other than gossipsub the capabilities of each "side" is not the same. For gossipsub each node acts as both a server and a client, whereas filter still has a server-side and client-side which needs to be differentiated.